### PR TITLE
Plane: use control_mode explicitly by index making the enum order arbitrary

### PR DESCRIPTION
**Pull request by magicrub**
This is a final cleanup to close https://github.com/ArduPilot/ardupilot/issues/4656

This is a non-functional change except for one tiny thing. Flight mode AUTOTUNE will no longer enforce a minimum ground speed which it was not supposed to be doing anyway. AUTOTUNE should be like FBWA which doesn't respect it (where FBWB would).

throttle_allows_nudging matches auto_navigation_mode in all cases except CIRCLE and Q* modes, and I don't think the throttle nudging does anything in Q but I wasn't sure. Anyway, a 1 bit-wide flag won't hurt.

### CODE REVIEW

File: ArduPlane/navigation.cpp [hunk 1]
```diff
@@ -127,15 +127,15 @@ void Plane::calc_airspeed_errors()
     // Set target to current airspeed + ground speed undershoot,
     // but only when this is faster than the target airspeed commanded
     // above.
-    if (control_mode >= FLY_BY_WIRE_B && (aparm.min_gndspeed_cm > 0)) {
+    if (auto_throttle_mode && (aparm.min_gndspeed_cm > 0)) {
```
Comment (WickedShell) [inline ArduPlane/navigation.cpp]: I think it's correct but this is a behavior change for CIRCLE.

Comment (magicrub) [reply ArduPlane/navigation.cpp]: as noted in the description. Although, CIRCLE would be pretty terrible if it wasn't making forward progress.. not much of a circle. Also, CIRCLE is mainly used in failsafe modes. 

This actually looks like I'm uncovering a bug. groundspeed_undershoot gets set only when there's a GPS.log >= 2D but never resets on GPS loss. The undershoot sticks forever. Same goes with if you change min_gndspeed_cm to zero during flight but at least its only used later when min_gndspeed_cm is >0.

Comment (WickedShell) [reply ArduPlane/navigation.cpp]: Circle doesn't actually draw a circle, it's actually a constant bank mode and doesn't care about drawing a circle on the ground. Given this thought I wonder if it really should just ignore the groundspeed. It's not actually maintaining a real circle, and is basically just a failsafe mode that shouldn't fall out of the sky but can wander wherever I think it's better that this maintain the airspeed rather then care about ground speed.

Comment (magicrub) [reply ArduPlane/navigation.cpp]: yup, fair enough. I'll put an exception in there for CIRCLE.

**Pull request merged**